docker/script/fetch_uniprot_data.py

```diff
@@ -0,0 +1,59 @@
+import requests
+import csv
+import argparse
+
+def fetch_uniprot_info(accession):
+    """Fetch UniProt taxonomy and proteome metadata for a given accession."""
+    url = f"https://rest.uniprot.org/uniprotkb/search?query=accession:{accession}&fields=xref_proteomes,organism_name&format=json"
+    response = requests.get(url)
+    data = response.json()
+    
+    try:
+        result = data['results'][0]
+        proteome_info = next(
+            (ref for ref in result['uniProtKBCrossReferences'] if ref['database'] == 'Proteomes'),
+            {}
+        )
+        proteome_id = proteome_info.get('id', '')
+        tax_id = result['organism']['taxonId']
+        lineage = result['organism']['lineage']
+        return {
+            "accession": accession,
+            "proteome_id": f"proteome-tax_id-{tax_id}-0_v4",
+            "tax_common_name": result['organism'].get('commonName', ''),
+            "tax_scientific_name": result['organism'].get('scientificName', ''),
+            "tax_lineage": ';'.join(lineage)
+        }
+    except (IndexError, KeyError) as e:
+        print(f"Failed to parse data for {accession}: {e}")
```

Comment:
If we are going to allow these processes to fail (i.e. catch errors and return empty data fields rather than dying loudly) then we should at least be consistent with how we represent "NULL" values in the output. 

Some other scripts are using the string "NA" to represent problematic data. This script is using an empty string.


Note: I would be tempted to change everything to use `pandas.NA`

```
import pandas as pd

MISSING_VALUE = pd.NA

...

        return {
            "accession": accession,
            "proteome_id": MISSING_VALUE,
            "tax_common_name": MISSING_VALUE,
            "tax_scientific_name": MISSING_VALUE,
            "tax_lineage": MISSING_VALUE
        }

```

https://pandas.pydata.org/docs/user_guide/missing_data.html